server/app/services/program/predicate/AndNode.java

```diff
@@ -40,7 +45,35 @@ public PredicateExpressionNodeType getType() {
 
   @Override
   public String toDisplayString(ImmutableList<QuestionDefinition> questions) {
+    // Sorted to ensure consistent rendering. A question should always be
+    // found for a child node but if it's not defaulting to Z prevents the
```

Comment:
I'm confused what is being protected against here.  I think you're hiding a coding error when questions are requested that don't actually exist. 

Coding errors shouldn't be hidden in general.

is there something else going on here?

Reply:
> Coding errors shouldn't be hidden in general.

Agree. I followed the same pattern here that was established in the preexisting nodes. I haven't investigated how big a change would be needed to improve it.

Reply:
I think I took this implication as worse than it is, it just effectively tries to order towards the end correct?  

Maybe call that out rather than specifying that the intent is to avoid an exception.

Reply:
done